packages/marko/src/core-tags/core/await/reorderer-renderer.js

```diff
@@ -54,13 +57,30 @@ module.exports = function (input, out) {
             global._afRuntime = true;
           }
 
-          asyncOut.write(
-            '<div id="af' +
-              awaitInfo.id +
-              '" style="display:none">' +
-              result.toString() +
-              "</div>"
-          );
+          if (global.cspNonce) {
+            asyncOut.write(
+              '<style nonce="' +
+                escapeDoubleQuotes(global.cspNonce) +
+                '">' +
+                "#af" +
+                awaitInfo.id +
+                "{display:none;}" +
+                "</style>" +
+                '<div id="af' +
+                awaitInfo.id +
+                '">' +
+                result.toString() +
+                "</div>"
+            );
+          } else {
+            asyncOut.write(
+              '<div id="af' +
+                awaitInfo.id +
+                '" style="display:none">' +
+                result.toString() +
+                "</div>"
```

Comment:
I opted to keep the original version, as this results in a shorter HTML if you don't need to worry about CSP. Happy to amend it so that it is consistent across both versions.

Reply:
Seems good to me 👍